File: python/asyncdemo_no_async.py
#!/usr/bin/env python3

import asyncio

@asyncio.coroutine
def return_number_async(n):
    print('in return_number_async')
    return n

@asyncio.coroutine
def silly_example():
    print('start of async function')
    fut = return_number_async(10)
    print('called return_number_async(10)')
    n = yield from fut
    print(f'result: {n}')

if __name__ == '__main__':
    loop = asyncio.new_event_loop()
    loop.run_until_complete(silly_example())

File: js/asyncdemo_no_async.js
#!/usr/bin/env node
"use strict"

function return_number_async(n) {
    return new Promise(function(resolve, reject) {
        console.log("in return_number_async")
        resolve(n)
    })
}

function silly_example() {
    console.log('start of async function')
    let fut = return_number_async(10)
    let my_promise = fut.then(function(n) {
        console.log(`result: ${n}`)
    })
    console.log('called return_number_async(10)')
    return my_promise
}

silly_example()
